packages/tpetra/CMakeLists.txt

```diff
@@ -21,6 +21,38 @@ TRIBITS_ADD_OPTION_AND_DEFINE(
   ${Tpetra_ENABLE_DEPRECATED_CODE_DEFAULT}
 )
 
+# Supported Kokkos version in Trilinos
+# NOTE: When we snapshot Kokkos into Trilinos, we have to update these numbers to maintain
+# compatibility with external Kokkos
+SET(Tpetra_SUPPORTED_KOKKOS_VERSION "4.0.0")
+
+
+# Option to allow developers to ignore incompatible Kokkos versions
+TRIBITS_ADD_OPTION_AND_DEFINE(
+  Tpetra_IGNORE_KOKKOS_COMPATIBILITY
+  TPETRA_IGNORE_KOKKOS_COMPATIBILITY
```

Comment:
What is the variable `TPETRA_IGNORE_KOKKOS_COMPATIBILITY` used for?  That second argument is typically used for a macro define in a `<Package>_Config.h.in` file that gets configured.   (See the documentation for [`tribits_add_option_and_define()`](https://tribitspub.github.io/TriBITS/users_guide/index.html#tribits-add-option-and-define).)

In this case, I think you just want:

```
set(TPETRA_IGNORE_KOKKOS_COMPATIBILITY OFF CACHE BOOL
  "If true, disables the Kokkos version compatibility checking in Tpetra")
```

or:

```
TRIBITS_ADD_OPTION_AND_DEFINE(
  Tpetra_IGNORE_KOKKOS_COMPATIBILITY
  ""
  "If true, disables the Kokkos version compatibility checking in Tpetra"
  OFF
)
```

if you want to export this variable `Tpetra_IGNORE_KOKKOS_COMPATIBILITY` to the `<Pacakge>Config.cmake` file (See [`tribits_pkg_export_cache_var()`](https://tribitspub.github.io/TriBITS/users_guide/index.html#tribits-pkg-export-cache-var).)